src/sparseml/recipe_template/cli.py

```diff
@@ -12,15 +14,61 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+import logging
+
+import click
 from sparseml import recipe_template
+from sparseml.version import __version__
+
+
+_LOGGER = logging.getLogger(__name__)
 
 
-def main(*args, **kwargs):
+@click.command(context_settings=(dict(show_default=True)))
+@click.version_option(version=__version__)
+@click.option(
+    "--pruning",
+    # other algos will be added in future
+    type=click.Choice(["true", "false", "gmp"], case_sensitive=False),
```

Comment:
what's the difference between `true` and `gmp`? do they use different algorithms? if so, recommend:
```
type=click.Choice(["gmp", <OTHER ALG>]),
default=None
```